activities.py

```diff
@@ -121,10 +121,14 @@ def entry_unique(self, spec_entry):
             e["title"].lower().strip() for e in self.data
         ]:
             raise ValueError(
-                ["%s already contains %s" % (self.filename, entry["title"])]
+                ["%s already contains title %s" % (self.filename, entry["title"])]
+            )
+        if entry["id"] in [e["id"] for e in self.data]:
+            raise ValueError(
+                ["%s already contains id %s" % (self.filename, entry["id"])]
             )
```

Comment:
Given the default value for `id` is `""`, I think it would be advisable to check for an empty ID here as well. While this code will prevent two such entries from validating, my reading is that it will allow _one_ to do so, which seems like it could become a recurring source of frustration.

Reply:
So the difficult thing about this is that while I could straightforwardly do this -- I'd then need to write code to make up an ID when adding the entry rather than adding an empty one, because the "add" code currently gives an error if the addition doesn't validate.  And if my code has edge cases where it doesn't work, people would have to fight with that code to get `./activities.py add` to work.

Reply:
Ugh. Okay, fair point.

Reply:
Actually, I guess I can just put the check *outside* `validate_entry`